src/schema/v2/me/showRecommendations.ts

```diff
@@ -0,0 +1,49 @@
+import { GraphQLFieldConfig } from "graphql"
+import { getPagingParameters, pageable } from "relay-cursor-paging"
+import { ResolverContext } from "types/graphql"
+import { connectionFromArraySlice } from "graphql-relay"
+import EventStatus from "../input_fields/event_status"
+import ShowSorts from "../sorts/show_sorts"
+import { ShowsConnection } from "../show"
+
+export const ShowRecommendations: GraphQLFieldConfig<void, ResolverContext> = {
+  type: ShowsConnection.connectionType,
+  args: pageable({
+    sort: {
+      type: ShowSorts,
+      defaultValue: "-created_at",
+    },
+    status: {
+      type: EventStatus.type,
+      defaultValue: "current",
+      description: "Filter shows by chronological event status",
+    },
+  }),
+  description: "A list of recommended shows for the user",
+  resolve: async (_root, options, { meShowsLoader }) => {
+    if (!meShowsLoader) return null
+
+    const { limit: size, offset } = getPagingParameters(options)
+    const { sort, status } = options
+
+    const gravityArgs = {
+      size,
+      offset,
+      total_count: true,
+      sort,
+      status,
+    }
+
+    const { body: shows, headers } = await meShowsLoader(gravityArgs)
+
+    const count = parseInt(headers["x-total-count"] || "0", 10)
+
+    return {
+      totalCount: count,
+      ...connectionFromArraySlice(shows, options, {
```

Comment:
The `return` here should be constructed using the `paginationResolver` per [these docs](https://github.com/artsy/metaphysics/blob/main/docs/pagination.md#4-resolver)

Also should include a comment (or in the description) that pagination doesn't actually work on this resolver (which is really not great tbh). Maybe this should be a `GraphQLList` type instead?

Also wondering about the name `showRecommendations`, I can see us building up _multiple_ such fields under `Me`, all of which are connections with the node type being a show. What if we have a more general `showsConnection` field under `Me`, that takes in a `mode: RECOMMENDATIONS` argument? That way it can be used for other ways a connection of shows is connected to `Me`. The edge fields could be enhanced as well with some meta information if needed as well (such as 'why' each show is included, or the time a show was 'followed', etc.). Not a blocker, but curious your thought on the design?

Reply:
didn't know about the helper as well